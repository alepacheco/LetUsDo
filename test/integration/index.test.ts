/* eslint-disable no-undef */
import { setup, teardown } from 'jest-dev-server';
import axios from 'axios';

describe('Integration tests', () => {
  beforeAll(async () => {
    jest.setTimeout(300000);
    await setup({
      command: `now dev --listen 9888`,
      launchTimeout: 300000,
      port: 9888
    });
    jest.setTimeout(5000);

    
  });

  afterAll(async () => {
    await teardown();
  });

  it('loads /api/', async () => {
    const { status, data } = await axios.get('http://localhost:9888/api/');

    expect(status).toBe(200);
    expect(data).toBe('Api index');
  });

  it('loads index.html page', async () => {
    const { status, data } = await axios.get('http://localhost:9888/');

    expect(status).toBe(200);
    expect(data.includes('<html>')).toEqual(true);
  });
});


