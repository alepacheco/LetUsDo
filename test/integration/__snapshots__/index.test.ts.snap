// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration tests loads index.html page 1`] = `"<!doctype html><html lang=\\"en\\"><head><link rel=\\"preconnect\\" href=\\"https://m.stripe.com\\"/><script async src=\\"https://js.stripe.com/v3/\\"></script><meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1,maximum-scale=1,user-scalable=no,shrink-to-fit=no\\"/><meta name=\\"HandheldFriendly\\" content=\\"true\\"/><meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\"/><meta charset=\\"utf-8\\"/><link rel=\\"stylesheet\\" href=\\"https://maxcdn.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css\\" integrity=\\"sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T\\" crossorigin=\\"anonymous\\"/><title>Let Us Do</title><link rel=\\"shortcut icon\\" href=\\"/favicon.ico\\"><link href=\\"/main.1e29b55669a7faafcd67.css\\" rel=\\"stylesheet\\"></head><body><div id=\\"app\\"></div><script type=\\"text/javascript\\" src=\\"/main.a8e43cf54a30ee28e206.js\\"></script></body></html>"`;
